components/ws-daemon/pkg/content/tar.go

```diff
@@ -86,7 +87,10 @@ func TarWithOptions(srcPath string, options *TarOptions) (io.ReadCloser, error)
 			}
 
 			if relFilePath != "." {
-				relFilePath = strings.Join([]string{".", relFilePath}, string(filepath.Separator))
+				buffer := bytes.NewBufferString(".")
```

Comment:
better? `filepath.Join`

Reply:
Not really
```shell
❯ go test -benchmem -run=^$ -bench . bench_test.go | prettybench
goos: linux
goarch: amd64
cpu: AMD Ryzen 9 5900X 12-Core Processor            
PASS
benchmark                                    iter     time/iter   bytes alloc        allocs
---------                                    ----     ---------   -----------        ------
BenchmarkNewBufferString-24              76683656   15.20 ns/op        0 B/op   0 allocs/op
BenchmarkNewBufferStringParallel/1-24   880934428    1.38 ns/op        0 B/op   0 allocs/op
BenchmarkNewBufferStringParallel/2-24   793567332    1.53 ns/op        0 B/op   0 allocs/op
BenchmarkNewBufferStringParallel/4-24   826919538    1.50 ns/op        0 B/op   0 allocs/op
BenchmarkNewBufferStringParallel/8-24   809302243    1.46 ns/op        0 B/op   0 allocs/op
BenchmarkFilePathJoin-24                 13993242   83.81 ns/op       40 B/op   3 allocs/op
BenchmarkFilePathJoinParallel/1-24       94575600   13.63 ns/op       40 B/op   3 allocs/op
BenchmarkFilePathJoinParallel/2-24       72334652   14.96 ns/op       40 B/op   3 allocs/op
BenchmarkFilePathJoinParallel/4-24       80168283   15.62 ns/op       40 B/op   3 allocs/op
BenchmarkFilePathJoinParallel/8-24       71974920   15.55 ns/op       40 B/op   3 allocs/op
ok  	command-line-arguments	12.709s
```

Reply:
To be clear, I am trying to tackle this issue
![Screenshot from 2021-08-10 21-24-54](https://user-images.githubusercontent.com/161571/129027271-b70e1c50-b85a-40c0-8845-1df11ae4a8e1.png)

Reply:
wow, that's surprising. Thank's for the benchmark 🚀